src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Base.Tests.cs

```diff
@@ -16,6 +16,16 @@ public abstract class RecommendCaseInsensitiveStringComparison_Base_Tests
 
         private static readonly string[] ContainsStartsWith = new[] { "Contains", "StartsWith" };
         private static readonly string[] UnnamedArgs = new[] { "", ", 1", ", 1, 1" };
+
+        private static readonly Tuple<string, string>[] CSharpComparisonOperators = new[] {
+            Tuple.Create("==", ""),
+            Tuple.Create("!=", "!")
```

Comment:
Shouldn't it cover `a.ToLower().Equals(b.ToLower())`, `!a.ToLower().Equals(b.ToLower())`, ... scenarios too?

Reply:
Maybe. It was not contemplated in the approved API proposal. I can ask in the main issue to determine if we want that case handled too. If that's the case, I would fix it in a separate PR.

Reply:
> Maybe. It was not contemplated in the approved API proposal. I can ask in the main issue to determine if we want that case handled too

string `==` operator just same as string.Equals and `!=` same as !string.Equals, in fact they actually calls string.Equals inside. https://github.com/dotnet/runtime/blob/0fce03e6bb0251f6e5d8abacd97d90f0a1a200f9/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs#L744-L746
I don't think its have to specifically mentioned in the API proposal